### ISSUE TITLE
Create outline for documentation around the core monorepo

### ISSUE THREAD

**Issue by desi** [MetaMask/contributor-docs#55]
This issue is for creating the outline of what would need to go into the contributor documenation around ideal package structure.

Please use the epic for inspiration on what to include here. [Epic 1588](https://github.com/MetaMask/core/issues/1588)

As part of this work please get feedback from team to finalize the outline and then create issues for filling in each of the areas of the outline. Please add the tickets to the epic and ensure they are pointed.

**Comment by mcmire** [comment]
Here's what I'm thinking for the outline:

1. Anatomy of the monorepo
   - Provide an ASCII version of the file tree structure, with the `packages/` directory collapsed, and a summary for each directory/file
1. Yarn
   - Why do we use Yarn?
   - Yarn "Berry"
   - Checking consistency of package manifests with Yarn constraints
1. TypeScript
   - What happens when you run `yarn build`
   - What are project references and how do they work?
   - Why is there a `tsconfig.json` and `tsconfig.build.json` in the root?
   - Why is there a `tsconfig.(build.)json` and also a `tsconfig.packages.(build.).json`?
   - How TypeScript resolves `@metamask/*` modules
1. Jest
   - What happens when you run `yarn test`
   - How Jest recognizes TypeScript code
   - How Jest resolves `@metamask/*` modules and how this differs from TypeScript
   - How code coverage is calculated, how to configure coverage thresholds, and how to exclude files/directories
1. ESLint
   - What happens when you run `yarn lint`
   - A note on ESLint overrides
1. TypeDoc
   - What happens when you run `yarn build:docs`
   - How TypeDoc is set up
1. How to update the dependency graphs in the README

Note that some of these topics could conceivably be split out in their own sections. We should try not to go into _too_ much detail, but mention enough to where people can understand which context/workflow a particular tool fits into.

**Comment by mcmire** [comment]
I agree we should have documentation about constraints. I'm curious whether that should be its own page, though, as I can envision that getting fairly large.
